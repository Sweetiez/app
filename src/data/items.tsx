import {ProductCard} from '../model';

export const fakeProducts: ProductCard[] = [
  {
    id: '1',
    name: 'Tartelette daim caramel',
    price: 10,
    images: [
      'https://picsum.photos/1200/300',
      'https://picsum.photos/1220/300',
      'https://picsum.photos/4000/1600',
    ],
    rating: 4,
    shortDescription: 'ma petite description',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
    comments: [
      {
        id: '1',
        author: 'Aymeric',
        content:
          'Tu vas voir, avec du safran ce sera parfait. Gusteau ne jure que par le safran.',
        rating: 4,
        date: '16/03/2022',
      },
      {
        id: '2',
        author: 'Gwendal',
        content:
          'La grande cuisine n’est pas faite pour les timorés. Il faut avoir de l’imagination, de l’audace. Il faut prendre le risque de commettre des erreurs et personne n’a le droit de vous imposer des limites, quelles que soient vos origines, d’où que vous veniez. Votre seule limite, c’est votre âme. C’est la vérité, tout le monde peut cuisiner, mais le véritable génie, n’appartient qu’aux audacieux.',
        rating: 3,
        date: '20/03/2022',
      },
      {
        id: '3',
        author: 'Antonin',
        content:
          'A bien des égards la tache du critique est aisée. Nous ne risquons pas grand chose, et pourtant nous jouissons d’une position de supériorité par rapport a ceux qui se soumettent avec leur travail à notre jugement. Nous nous épanouissons dans la critique négative, plaisante à écrire et à lire. Mais l’amère vérité qu’il faut bien regarder en face c’est que, dans le grand ordre des choses, le met le plus médiocre a sans doute plus de valeur que notre critique qui le dénonce comme tel. Il est pourtant des circonstances où le critique prend un vrai risque. C’est lorsqu’il découvre et défend l’innovation. Le monde est souvent malveillant à l’encontre des nouveaux talents et de la création. Le nouveau a besoin d’amis. Hier soir j’ai vécu une expérience inédite et dégusté un plat extraordinaire d’une origine singulière s’il en est. Avancer que ce plat et son créateur ont radicalement changé l’idée que je me faisais de la grande cuisine serait peu dire. Ils m’ont bouleversé au plus profond de mon être. Je n’ai jamais fait mystère du mépris que m’inspirait la devise d’Auguste Gusteau « Tout le monde peut cuisiner ! ». Mais ce n’est qu’aujourd’hui, aujourd’hui seulement que je comprends vraiment ce qu’il voulait dire. Tout le monde ne peut pas devenir un grand artiste, mais un grand artiste peut surgir n’importe où. Il est difficile d’imaginer origine plus modeste que celle du génie qui officie maintenant chez Gusteau et qui est a nos yeux, rien moins que le plus grand cuisinier de France. Je retournerai bientôt chez Gusteau… plus affamé que jamais.',
        rating: 5,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '2',
    name: 'Tartelete Citron meringué',
    price: 10,
    images: ['https://picsum.photos/1220/300'],
    rating: 4,
    shortDescription:
      'large description large description large description large description large description large description large description large description large description',
    description: 'This is a test description',
    comments: [
      {
        id: '2',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '20/03/2022',
      },
    ],
  },
  {
    id: '3',
    name: 'Mi-cuit chocolat',
    price: 10,
    images: ['https://picsum.photos/200/500'],
    rating: 4,
    shortDescription:
      'ma moyenne description ma moyenne description ma moyenne description',
    description:
      'This is a test large description This is a test large description',
    comments: [
      {
        id: '3',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '4',
    name: 'Crème brûlée',
    price: 10,
    images: ['https://picsum.photos/200/600'],
    rating: 5,
    shortDescription: 'ma petite description',
    description: 'This is a test description',
    comments: [
      {
        id: '4',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '5',
    name: 'Product 5',
    price: 10,
    images: ['https://picsum.photos/200/700'],
    rating: 1,
    shortDescription: 'ma petite description',
    description: 'This is a test description',
    comments: [
      {
        id: '5',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '6',
    name: 'Product 6',
    price: 10,
    images: ['https://picsum.photos/200/800'],
    rating: 2,
    shortDescription: 'ma petite description',
    description: 'This is a test description',
    comments: [
      {
        id: '6',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '7',
    name: 'Product 7',
    price: 10,
    shortDescription: 'ma petite description',
    images: ['https://picsum.photos/200/900'],
    rating: 3,
    description: 'This is a test description',
    comments: [
      {
        id: '7',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '8',
    name: 'Product 8',
    price: 10,
    images: ['https://picsum.photos/200/300'],
    rating: 3.3,
    shortDescription: 'ma petite description',
    description: 'This is a test description',
    comments: [
      {
        id: '8',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '9',
    name: 'Product 9',
    price: 10,
    images: ['https://picsum.photos/200/300'],
    rating: 4,
    shortDescription: 'ma petite description',
    description: 'This is a test description',
    comments: [
      {
        id: '9',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '10',
    name: 'Product 10',
    price: 10,
    images: ['https://picsum.photos/200/200'],
    rating: 4.5,
    shortDescription: 'ma petite description',
    description: 'This is a test description',
    comments: [
      {
        id: '10',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '11',
    name: 'Product 11',
    price: 10,
    images: ['https://picsum.photos/100/300'],
    rating: 0,
    shortDescription: 'ma petite description',
    description: 'This is a test description',
    comments: [
      {
        id: '11',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '12',
    name: 'Product 12',
    price: 10,
    images: ['https://picsum.photos/300/300'],
    rating: 1,
    shortDescription: 'ma petite description',
    description: 'This is a test description',
    comments: [
      {
        id: '12',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
  {
    id: '13',
    name: 'Product 13',
    price: 10,
    images: ['https://picsum.photos/400/300'],
    rating: 2,
    shortDescription: 'ma petite description',
    description: 'This is a test description',
    comments: [
      {
        id: '13',
        author: 'Aymeric',
        content: 'Miam',
        rating: 4,
        date: '27/03/2022',
      },
    ],
  },
];
