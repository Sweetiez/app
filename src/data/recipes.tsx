import {Recipe} from '../model';

export const fakeRecipes: Recipe[] = [
  {
    id: '1',
    name: 'Tartelette daim caramel',
    images: [
      'https://picsum.photos/4000/1600',
      'https://picsum.photos/4000/1600',
    ],
    rating: 3,
    description:
      'Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
    totalTime: '30',
    preparationTime: '15',
    sleepTime: '5',
    cookingTime: '10',
    level: 'Débutant',
    cost: 50,
    portions: 4,
    steps: [
      {
        id: '1',
        number: 1,
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
      },
      {
        id: '2',
        number: 2,
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
      },
      {
        id: '3',
        number: 3,
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
      },
    ],
  },
  {
    id: '2',
    name: 'Tartelete Citron meringué',
    images: ['https://picsum.photos/4000/1600'],
    rating: 5,
    description:
      'Sahhh ... une masterclass !! Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
    totalTime: '30',
    preparationTime: '15',
    sleepTime: '5',
    cookingTime: '10',
    level: 'Débutant',
    cost: 50,
    portions: 4,
    steps: [
      {
        id: '1',
        number: 1,
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
      },
    ],
  },
  {
    id: '3',
    name: 'Mini burgers',
    images: ['https://picsum.photos/4000/1600'],
    rating: 5,
    description:
      'Sahhh ... une masterclass !! Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
    totalTime: '30',
    preparationTime: '15',
    sleepTime: '5',
    cookingTime: '10',
    level: 'Normal',
    cost: 50,
    portions: 4,
    steps: [
      {
        id: '1',
        number: 1,
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
      },
    ],
  },
  {
    id: '4',
    name: 'Jai vraiment faim',
    images: ['https://picsum.photos/4000/1600'],
    rating: 0,
    description:
      'laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
    totalTime: '30',
    preparationTime: '15',
    sleepTime: '5',
    cookingTime: '10',
    level: 'Difficile',
    cost: 50,
    portions: 4,
    steps: [
      {
        id: '1',
        number: 1,
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?',
      },
    ],
  },
];
