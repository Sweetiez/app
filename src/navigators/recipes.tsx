import React from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {AccountScreen} from '../pages';

const RecipesStack = createNativeStackNavigator();

function RecipesStackScreen() {
  return (
    <RecipesStack.Navigator screenOptions={{headerShown: false}}>
      <RecipesStack.Screen name="Account" component={AccountScreen} />
    </RecipesStack.Navigator>
  );
}

export default RecipesStackScreen;
