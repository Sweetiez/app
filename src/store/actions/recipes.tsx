import {UPDATE_RECIPES} from '../constants';

export function updateRecipes(recipes) {
  return {
    type: UPDATE_RECIPES,
    payload: recipes,
  };
}
