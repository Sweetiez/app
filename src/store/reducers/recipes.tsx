import {UPDATE_RECIPES} from '../constants';

const initialState = {
  recipes: [],
};
const shopReducer = (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_RECIPES:
      return {
        ...state,
        recipes: action.payload,
      };
    default:
      return state;
  }
};
export default shopReducer;
