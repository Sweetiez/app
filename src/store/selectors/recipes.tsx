import State from '../../model/store-model';

export default function recipesSelector(state: State) {
  return state.recipes.recipes;
}
