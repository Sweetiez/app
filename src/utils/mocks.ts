jest.mock('./../assets/icons/pen.svg', () => 'Pen');
jest.mock('./../assets/icons/star.svg', () => 'Star');
jest.mock('./../assets/icons/plus.svg', () => 'Plus');
jest.mock('./../assets/icons/minus.svg', () => 'Minus');
jest.mock('./../assets/icons/add.svg', () => 'Add');
jest.mock('./../assets/icons/arrow-back.svg', () => 'Back');
jest.mock('./../assets/icons/cook-book.svg', () => 'CookBook');
jest.mock('./../assets/icons/cake.svg', () => 'Cake');
jest.mock('./../assets/icons/event.svg', () => 'Event');
jest.mock('./../assets/icons/eye.svg', () => 'Eye');
jest.mock('./../assets/icons/trash.svg', () => 'Trash');
jest.mock('./../assets/icons/no-image.svg', () => 'NoImage');
jest.mock('./../assets/icons/edit.svg', () => 'Edit');
jest.mock('./../assets/icons/warning.svg', () => 'Warning');

jest.mock('react-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}))



jest.mock('react-native-picker-select', () => 'RNPickerSelect')
jest.mock('react-native-config', () => 'Config')