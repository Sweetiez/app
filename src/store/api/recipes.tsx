import {RECIPES_ERROR, RECIPE_ERROR} from '../constants';
import {buildRequest} from '../../utils/api';

export function getRecipes() {
  return buildRequest(
    'GET',
    '/recipes',
    undefined,
    () => {
      return RECIPES_ERROR;
    },
    response => {
      return response.json();
    },
    undefined,
  );
}

export function getRecipe(id) {
  return buildRequest(
    'GET',
    '/recipes/' + id,
    undefined,
    () => {
      return RECIPE_ERROR;
    },
    response => {
      return response.json();
    },
    undefined,
  );
}
