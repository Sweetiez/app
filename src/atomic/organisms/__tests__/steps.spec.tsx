import React from 'react';
import renderer from 'react-test-renderer';
import '../../../utils/mocks';
import Steps from '../steps';
import {fakeRecipes} from '../../../data/recipes';

describe('Steps', () => {
  it('renders correctly', () => {
    const steps = fakeRecipes[0].steps;
    const tree = renderer.create(<Steps steps={steps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
