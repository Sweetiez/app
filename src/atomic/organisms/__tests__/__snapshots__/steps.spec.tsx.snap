// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 0,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 0,
        "paddingTop": 10,
        "shadowColor": "rgba(0, 0, 0, 0.07)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
      },
    ]
  }
>
  <Text
    color="#dba970"
    size={28}
    style={
      Array [
        Object {
          "color": "#dba970",
          "fontFamily": "Pompiere-Regular",
          "fontSize": 28,
        },
      ]
    }
  >
    recipe.steps
  </Text>
  <View
    style={
      Array [
        Object {
          "marginBottom": 10,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginBottom": 5,
          "paddingRight": 20,
        },
      ]
    }
  >
    <Text
      color="#dba970"
      size={16}
      style={
        Array [
          Object {
            "color": "#dba970",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      1 . 
    </Text>
    <Text
      color="#000000"
      size={16}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginBottom": 5,
          "paddingRight": 20,
        },
      ]
    }
  >
    <Text
      color="#dba970"
      size={16}
      style={
        Array [
          Object {
            "color": "#dba970",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      2 . 
    </Text>
    <Text
      color="#000000"
      size={16}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginBottom": 5,
          "paddingRight": 20,
        },
      ]
    }
  >
    <Text
      color="#dba970"
      size={16}
      style={
        Array [
          Object {
            "color": "#dba970",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      3 . 
    </Text>
    <Text
      color="#000000"
      size={16}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. A adipisci alias aliquam culpa, delectus expedita fugit ipsum laboriosam maiores maxime numquam reprehenderit, rerum sint unde vero! Accusamus id quod repudiandae?
    </Text>
  </View>
</View>
`;
